var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -162,6 +171,40 @@ def install(self, spec, prefix):
         else:
             configure_args.append("--disable-debug")
 
+        if '+cuda' in self.spec:
+            configure_args.extend([
+                '--with-cuda',
+                '--enable-curand',
```

Comment:
Does it make sense for the user to enable/disable one or more of `cuda`, `curand`, and `cub` or are the three variants tied together as it appears with this if-statement?

Reply:
@sayerhs do you know?

Reply:
@eugeneswalker This is how I'd implemented it for hypre+cuda. So cub/curand are only activated if the user adds it to hypre spec.

```python
        configure_args.append(
            '--%s-cuda'%('with' if '+cuda' in self.spec else 'without'))
        if '+cuda' in self.spec:
            configure_args.extend([
                self._cfg_opt_from_spec('curand'),
                self._cfg_opt_from_spec('cub'),
            ])

        if '~cuda+int64' in self.spec:
            configure_args.append('--enable-bigint')
        else:
            configure_args.append('--disable-bigint')
```